src/quantity-selector/index.ts

```diff
@@ -0,0 +1,68 @@
+import { alias, tag, Events, Selectors, Tag } from '@storefront/core';
+
+const INITIAL_INDEX = 0;
+
+@alias('selector')
+@tag('gb-quantity-selector', require('./index.html'))
+class QuantitySelector {
+
+  state: QuantitySelector.State = {
+    selected: INITIAL_INDEX,
+    options: []
+  };
+
+  init() {
+    this.getOptions(this.props.quantity);
+
+    // args for this event is a url link
+    this.flux.on(Events.URL_UPDATED, this.reset);
+  }
+
+  onMount() {
+    this.flux.emit('selector:update', this.state.options[this.state.selected].value);
+  }
+
+  getOptions = (quantity: number) => {
+    let options = [];
+    let i;
+    for (i = 0; i < quantity; i++) {
+      let el = {};
+      const value = i + 1;
+      el = { value };
+      el['label'] = value.toString();
+      el['selected'] = (i === this.state.selected ? true : false);
+      options.push(el);
+    }
+    this.state.options = options;
+  }
+
+  setSelected = (target: number = INITIAL_INDEX) => {
+    this.state.selected = target;
+    this.state.options.forEach((el, i) => {
+      el['selected'] = (i === this.state.selected ? true : false);
+    });
+
+    this.flux.emit('selector:update', this.state.options[this.state.selected].value);
```

Comment:
I think the event name could be better, 'selector:update' does not really say much about quantity-selector

Reply:
could you think of anything? I am following the lead in autocomplete

Reply:
How about 'cart-quantity-selector:update' just to be clear